src/sage/rings/species.py

```diff
@@ -0,0 +1,2341 @@
+                sage: len(P.cover_relations())
+                17
+                sage: sorted(P.cover_relations(), key=str)
+                [[C_4, {((1,2)(3,4),)}],
+                 [E_2^2, X^2*E_2],
+                 [E_2^2, {((1,2)(3,4),)}],
+                 [E_4, Eo_4],
+                 [E_4, P_4],
+                 [E_4, X*E_3],
+                 [Eo_4, Pb_4],
+                 [Eo_4, X*C_3],
+                 [P_4, C_4],
+                 [P_4, E_2^2],
+                 [P_4, Pb_4],
+                 [Pb_4, {((1,2)(3,4),)}],
+                 [X*C_3, X^4],
+                 [X*E_3, X*C_3],
+                 [X*E_3, X^2*E_2],
+                 [X^2*E_2, X^4],
+                 [{((1,2)(3,4),)}, X^4]]
+
+            TESTS::
+
+                sage: [(a, b) for a, b in Subsets(M.subset(4), 2) if (a < b) != (b > a)]
+                []
+                sage: [(a, b) for a, b in Subsets(M.subset(4), 2) if (a <= b) != (b >= a)]
+                []
+
+                sage: M = MolecularSpecies("S, T")
+                sage: S = M(SymmetricGroup(1), {0: [1]})
+                sage: T = M(SymmetricGroup(1), {1: [1]})
+                sage: E2S = M(SymmetricGroup(2), {0: [1,2]})
+                sage: T * E2S < S^2 * T
+                True
+            """
+            if op is op_EQ or op is op_NE:
+                return super()._richcmp_(other, op)
+            if op is op_LE:
+                return self == other or self < other
+            if op is op_GE:
+                return other <= self
+            if op is op_GT:
+                return other < self
+            if op is op_LT:
+                # the arities match because the parents are equal
+                if self.grade() != other.grade():
+                    # X should come before Y
+                    return (sum(self.grade()) < sum(other.grade())
+                            or (sum(self.grade()) == sum(other.grade())
+                                and self.grade() > other.grade()))
+
+                S = SymmetricGroup(sum(self.grade())).young_subgroup(self.grade())
+                # conjugate self and other to match S
+                G, G_dompart = self.permutation_group()
+                g = list(chain.from_iterable(G_dompart))
+                conj_self = PermutationGroupElement(g).inverse()
+                G = libgap.ConjugateGroup(G, conj_self)
+                H, H_dompart = other.permutation_group()
+                h = list(chain.from_iterable(H_dompart))
+                conj_other = PermutationGroupElement(h).inverse()
+                H = libgap.ConjugateGroup(H, conj_other)
+                return GAP_FAIL != libgap.ContainedConjugates(S, G, H, True)
+
+        @cached_method
+        def permutation_group(self):
+            r"""
+            Return the (transitive) permutation group
+            corresponding to ``self``, together with the partition of
+            the domain into sorts.
```

Comment:
What is a `sort`? Did you mean `sets`?

Reply:
Sort is species language for variable (which might be misleading).  I added some documentation at module level and changed text to consistently use sort.